compiler/rustc_typeck/src/structured_errors/missing_cast_for_variadic_arg.rs

```diff
@@ -32,12 +32,21 @@ impl<'tcx> StructuredDiagnostic<'tcx> for MissingCastForVariadicArg<'tcx> {
         }
 
         if let Ok(snippet) = self.sess.source_map().span_to_snippet(self.span) {
-            err.span_suggestion(
-                self.span,
-                &format!("cast the value to `{}`", self.cast_ty),
-                format!("{} as {}", snippet, self.cast_ty),
-                Applicability::MachineApplicable,
-            );
+            if self.ty.is_fn() {
+                err.span_suggestion(
```

Comment:
It might be useful here to link to something like: https://doc.rust-lang.org/reference/types/function-item.html

Or at least note that a function item type is zero-sized, and needs to be casted to a pointer to be used in FFI.